components/dispatcher/internal/dispatcherhealth/dispatcherHealth.go

```diff
@@ -0,0 +1,58 @@
+package dispatcherhealth
+
+import (
+	"github.com/kyma-incubator/knative-kafka/components/common/pkg/kafka/health"
+	"sync"
+)
+
+// Start The HTTP Server Listening For Requests
+
+type Server struct {
+	health.Server
+
+	// Additional Synchronization Mutexes
+	dispatcherMutex sync.Mutex // Synchronizes access to the dispatcherReady flag
+
+	// Additional Internal Flags
+	dispatcherReady bool // A flag that the producer sets when it is ready
+}
+
+// Creates A New Server With Specified Configuration
+func NewDispatcherHealthServer(httpPort string) *Server {
+	dispatcherHealth := &Server{}
+	health := health.NewHealthServer(httpPort, dispatcherHealth)
+	dispatcherHealth.Server = *health
+
+	// Return The Server
+	return dispatcherHealth
+}
+
+// Synchronized Function To Set Producer Ready Flag
+func (chs *Server) SetDispatcherReady(isReady bool) {
+	chs.dispatcherMutex.Lock()
+	chs.dispatcherReady = isReady
+	chs.dispatcherMutex.Unlock()
+}
+
+// Set All Liveness And Readiness Flags To False
+func (chs *Server) ShuttingDown() {
+	chs.Server.ShuttingDown()
+	chs.SetDispatcherReady(false)
+}
+
+// Access Function For DispatcherReady Flag
+func (chs *Server) IsDispatcherReady() bool {
+	return chs.dispatcherReady
```

Comment:
As mentioned above shouldn't use Is in a getter name in go.  Also with IsReady and IsAlive

Reply:
Altered as suggested